### DISCUSSION TITLE
Is there a way to have an image rendered inline with other text?

### DISCUSSION THREAD

**Discussion by hemna** [AnonymouX47/term-image#145] [General]
Hi I'm playing around with this lib and trying to do something like this

from term_image.image import from_file

img = from_file("/path_to_my/icon.png")
img.height=1
print(f"Hello there, my image is right here ----> {image}")

Although this "works", but the rendering looks like this, where the image occupies the center and entire contents of the visible
terminal.

```
Hello there, my image is right here ---->









                                                                                                 <image here>







```

**Comment by AnonymouX47** [comment]
Hello. :smiley:
Sorry for the late reply. :pray:

Yes, it's possible with:

```py
print(f"Hello there, my image is right here ----> {image:1.1}")
```

which specifies padding width and height each to be 1 cell, thereby ensuring there's no padding around the image.

See the [Render Format Specification](https://term-image.readthedocs.io/en/v0.7.2/guide/formatting.html#render-format-specification) and [examples](https://term-image.readthedocs.io/en/v0.7.2/start/tutorial.html#formatted-render).

You're getting the output reported due to the _default_ **padding** applied by the `__format__()` method.

Sorry for the clumsiness. A new, neater and more intuitive API is already in the works.

Thank you very much. :smiley:
